src/renderer/components/opds/PageNavigation.tsx

```diff
@@ -0,0 +1,85 @@
+// ==LICENSE-BEGIN==
+// Copyright 2017 European Digital Reading Lab. All rights reserved.
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as React from "react";
+
+import { TranslatorProps, withTranslator } from "readium-desktop/renderer/components/utils/hoc/translator";
+
+import * as ArrowRightIcon from "readium-desktop/renderer/assets/icons/baseline-arrow_forward_ios-24px.svg";
+import * as ArrowLeftIcon from "readium-desktop/renderer/assets/icons/baseline-arrow_left_ios-24px.svg";
+import SVG from "readium-desktop/renderer/components/utils/SVG";
+
+import { OpdsResultUrls } from "readium-desktop/common/views/opds";
+
+import * as styles from "readium-desktop/renderer/assets/styles/opds.css";
+
+interface Props extends TranslatorProps {
+    goto: (url: string) => void;
+    urls: OpdsResultUrls;
+}
+
+class PageNavigation extends React.Component<Props> {
+    public constructor(props: Props) {
+        super(props);
+
+        this.handleKeyDown = this.handleKeyDown.bind(this);
+    }
+
+    public componentDidMount() {
+        document.addEventListener("keydown", this.handleKeyDown);
+    }
+
+    public componentWillUnmount() {
+        document.removeEventListener("keydown", this.handleKeyDown);
+    }
+
+    public render() {
+        const { goto, urls, __ } = this.props;
+        const { nextPage, previousPage, lastPage, firstPage } = urls;
+        return (
+            <div className={styles.opds_page_navigation}>
+                { firstPage ?
+                    <button onClick={() => goto(firstPage)}>
+                        {__("opds.firstPage")}
+                    </button>
+                : <div/> }
+                { previousPage ?
+                    <button onClick={() => goto(previousPage)}>
+                        <SVG svg={ArrowLeftIcon} />
+                        {__("opds.previous")}
+                    </button>
+                : <div/> }
+                { nextPage ?
+                    <button onClick={() => goto(nextPage)}>
+                        {__("opds.next")}
+                        <SVG svg={ArrowRightIcon} />
+                    </button>
+                : <div/>}
+                { lastPage ?
+                    <button onClick={() => goto(lastPage)}>
+                        {__("opds.lastPage")}
+                    </button>
+                : <div/> }
+            </div>
+        );
+    }
+
+    private handleKeyDown(e: KeyboardEvent) {
```

Comment:
Because `keydown` events can fired in rapid succession, this can be problematic with the async/promise calls underlying to the implementation of `goto()` (`props.browse()`). Therefore, I think calls to `handleKeyDown()` should be throttled (not debounced).

Reply:
I did it in my last commit : navigation shortkeys can be used every 2 seconds or when the page is refreshed